### PR TITLE
plasmidfinder data manager V1

### PR DESCRIPTION
**Pull request by pimarin**
Data manager for the plasmidfinder tool

- It only update from bitbucket repository the latest version and keep the commit number in name

### CODE REVIEW

File: data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<tool id="data_manager_fetch_plasmidfinder" name="plasmidfinder_datamanager" tool_type="manage_data" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description>AMRfinderplus database builder</description>
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.xml:2]: description is probably wrong.

File: data_managers/data_manager_fetch_plasmidfinder/plasmidfinder_database.loc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# this is a tab separated file describing the location of plasmidfinder database
+# The name was obtained by merging date of download and commit number
+# the columns are:
+# value, name, date, path
+#
+# for example
+#plasmidfinder_9002e7282dd0_2022-12-19    9002e7282dd0_2022-12-19    plasmidfinder-db
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/plasmidfinder_database.loc:7]: The example line has only 3 columns?

File: data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+class DownloadPlasmidfinderDatabase(GetPlasmidfinderDataManager):
+    """
+    Download the amrfinderplus database from the ncbi.
+    Make the database available with hmm and indexed files
+    Build the data manager infos for galaxy
+    """
+
+    def __init__(self,
+                 output_dir=Path.cwd(),
+                 plasmidfinder_url="https://bitbucket.org/genomicepidemiology/plasmidfinder_db/get/master.gz",
```
Comment (bernt-matthias) [inline data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.py:66]: Wondering if the authors of the DB could do releases (or do already https://bitbucket.org/genomicepidemiology/plasmidfinder_db/downloads/?tab=tags)?

File: data_managers/data_manager_fetch_plasmidfinder/plasmidfinder_database.loc (hunk 2 above)
Comment (bernt-matthias) [reply data_managers/data_manager_fetch_plasmidfinder/plasmidfinder_database.loc:7]: Is the commit number helpful?

File: data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.py (hunk 3 above)
Comment (pimarin) [reply data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.py:66]: The release date is 2019 and last update of the master is 2022 by adding more plasmid sequences...
I can add an option with "latest" or release number.

Comment (bernt-matthias) [reply data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.py:66]: Maybe ask them (upstream) if they could make releases?

Comment (pimarin) [reply data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.py:66]: I asked [here](https://bitbucket.org/genomicepidemiology/plasmidfinder_db/issues/3/add-release-version).
But awaiting the answer, I make a modification to suggest latest or '2.1'(the only release version). And i could be easy to add new version by modifying only 1 method of the script

**Comment by pimarin** [comment]
I wondering if it possible to simplify my tests with only 1 json file for 2 different outputs ?

**Comment by pimarin** [comment]
I update the version of the database depending if a release is available, if 'latest' version is required, the commit value is added in the name of the database. 
I think for reproducibility, is more easy to reuse the good version of the palsmidfinder database if we provide the commit

**Comment by pimarin** [comment]
is it okay ?

**Review by bgruening — APPROVED** [review]

File: data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.xml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    <outputs>
+        <data name="output_file" format="data_manager_json"/>
+    </outputs>
+    <tests>
+        <!-- Test_1 DB latest -->
+        <test expect_num_outputs="1">
+            <param name="database_select" value="latest"/>
+            <output name="output_file" value="plasmidfinder_test_data_manager_latest.json">
+                <assert_contents>
+                    <has_text_matching expression='{"data_tables": {"plasmidfinder_database": [{"date": "\d\d\d\d-\d\d-\d\d", "name": "b1c9ddac0e61_\d\d\d\d-\d\d-\d\d", "path": "plasmidfinder-db", "value": "plasmidfinder_b1c9ddac0e61_\d\d\d\d-\d\d-\d\d"' />
```
Comment (bgruening) [inline data_managers/data_manager_fetch_plasmidfinder/data_manager/plasmidfinder_fetch_database.xml:28]: I think we need to adjust the test as the date is included and will fail always. I tried to use has_text_matching here, but you probably need to adjust this a bit.

**Pull request closed without merging**